draft-ietf-tls-tls13.md

```diff
@@ -3428,18 +3428,17 @@ mechanism through TLS extensions or some other means.
 
 ## Limits on Key Usage
 
-There are cryptographic limits on the amount of plaintext which can be
-safely encrypted under a given set of keys.  {{AEAD-LIMITS}} provides
-an analysis of these limits under the assumption that the underlying
-primitive (AES or ChaCha20) has no weaknesses. Implementations SHOULD
-do a key update {{key-update}} prior to reaching these limits.
-
-For AES-GCM, up to 2^24.5 full-size records (about 24 million)
-may be encrypted on a
-given connection while keeping a safety margin of approximately
-2^-57 for Authenticated Encryption (AE) security. For
-ChaCha20/Poly1305, the record sequence number would wrap before the
-safety limit is reached.
+Using AES-GCM to provide authenticity of authenticated data, content of the plaintext and information leakage (see Note below) protection for the plaintext, the maximum amount of data (padded records) which can be safely encrypted with a single key is 2^48 128-bit blocks (2^64 bytes). When the maximum amount of data is reached, the chance of having a collision among 128-bit blocks of the ciphertext is below 1^(-32) which is negligible. The recommended maximum amount of data (2^64 bytes) is made with the assumption that the (padded) record size is either 128 bits, a multiple of 128 bits or a lot larger than 128 bits. If the (padded) record size is 2^x bytes, then the data limit with GCM for each key is 2^(64-x) records.
```

Comment:
My calculation is correct as I explained in the email. 1^(-32) was a typo, it should have been 2^(-32).

Why did you say that the last sentence is incorrect?

When padded record size is much larger than 128 bits (one block), then the few bits padding done by AES (due to the partial last block) is negligible comparing to the record size. 

When a (padded or not) record is less than 1 AES block, then the data limit is reduced to the ration of the x (bits) over 128 because each AES encryption produces 128-bit output.